cpp/include/cugraph/algorithms.hpp

```diff
@@ -779,6 +781,51 @@ void ecg(raft::handle_t const& handle,
          vertex_t ensemble_size,
          vertex_t* clustering);
 
+/**
+ * @brief Computes the ecg clustering of the given graph.
+ *
+ * ECG runs truncated Louvain on an ensemble of permutations of the input graph,
+ * then uses the ensemble partitions to determine weights for the input graph.
+ * The final result is found by running full Louvain on the input graph using
+ * the determined weights. See https://arxiv.org/abs/1809.05578 for further
+ * information.
+ *
+ * @throws     cugraph::logic_error when an error occurs.
+ *
```

Comment:
Missing documentation for template parameters (and could you fix the documentation error for Louvain as well? Louvain is using the same template parameters but it's tparam documentation is listing graph_view_t).
``` 
 * @tparam vertex_t Type of vertex identifiers. Needs to be an integral type.
 * @tparam edge_t Type of edge identifiers. Needs to be an integral type.
 * @tparam weight_t Type of edge weights. Needs to be a floating point type.
 * @tparam multi_gpu Flag indicating whether template instantiation should target single-GPU (false)
 ```